decidim-core/lib/decidim/content_parsers/base_parser.rb

```diff
@@ -0,0 +1,29 @@
+# frozen_string_literal: true
+
+module Decidim
+  module ContentParsers
+    class BaseParser
+      attr_reader :content
+
+      # Gets initialized with the `content` to parse. It can either receive
+      # already rewritten content or regular content.
+      def initialize(content)
+        @content = content || ""
+      end
+
+      # Parse the the `content` and return it modified
+      def rewrite
+        raise NotImplementedError
```

Comment:
Do not use `NotImplementedError`, as this is not its use case:

http://chrisstump.online/2016/03/23/stop-abusing-notimplementederror/

You cann `raise` with a string saying it should be defined, for example

Reply:
It is always good to read articles like this, thank you. I'm refactoring to this:

```
def rewrite
  raise "#{self.class.name} is expected to implement #rewrite"
end
```

Also note that the `NotImplementedError` is used in other places of the code for this use case, so there is room to improve the base code. I do not have time right now to refactor the other parts. Perhaps we can open an issue as a `good first issue`, it's okay? Or can you take care of this?

Reply:
Open an issue please 😄